src/planner/ompl/Planner.cpp

```diff
@@ -317,6 +317,68 @@ trajectory::InterpolatedPtr planCRRTConnect(
                   std::move(_interpolator), _maxPlanTime);
 }
 
+//=============================================================================
+trajectory::InterpolatedPtr simplifyOMPL(statespace::StateSpacePtr _stateSpace,
+                                           statespace::InterpolatorPtr _interpolator,
+                                           distance::DistanceMetricPtr _dmetric,
+                                           constraint::SampleablePtr _sampler,
+                                           constraint::TestablePtr _validityConstraint,
+                                           constraint::TestablePtr _boundsConstraint,
+                                           constraint::ProjectablePtr _boundsProjector, 
+                                           double _maxDistanceBtwValidityChecks,
+                                           trajectory::InterpolatedPtr _originalTraj)
+{
+
+// Step 1: Generate the space information of OMPL type
+auto si = getSpaceInformation(
+      _stateSpace, _interpolator, std::move(_dmetric), std::move(_sampler),
+      std::move(_validityConstraint), std::move(_boundsConstraint),
+      std::move(_boundsProjector), _maxDistanceBtwValidityChecks);
+
+// Get the state space
+auto sspace = ompl_static_pointer_cast<GeometricStateSpace>(
+      si->getStateSpace());
+
+
+// Step 2: Convert the AIKIDO trajectory to Geometric Path 
+::ompl::geometric::PathGeometric *path = new ::ompl::geometric::PathGeometric(si);
+
+for (size_t idx = 0; idx < _originalTraj->getNumWaypoints(); ++idx) 
+{
+      auto ompl_state = sspace->allocState(_originalTraj->getWaypoint(idx));
+      path->append(ompl_state);
+      sspace->freeState(ompl_state);
+}
+
+// Step 3: Use the OMPL methods to simplify the path
+::ompl::geometric::PathSimplifierPtr simplifier;
+simplifier.reset(new ::ompl::geometric::PathSimplifier(si));
+bool const shortened = simplifier->shortcutPath(*path, 1, 1, 1.0, 0.005); 
```

Comment:
`maxEmptySteps` seems to be an intuitive and natural reason for terminating the shortcutting procedure. So shall keep a count of empty steps through iterations and also use a timeout to bound time of shortcutting from top.